[Dynamo] graph break on isinstance calls if we don't know the type

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #118928
* __->__ #118778
* #118768

If we can't figure out the python type of a VariableTracker, then the
isinstance call should graph break (instead of raising an error).

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

zou3519 committed:
[Dynamo] graph break on isinstance calls if we don't know the type

If we can't figure out the python type of a VariableTracker, then the
isinstance call should graph break (instead of raising an error).

[ghstack-poisoned]

zou3519 committed:
Update on "[Dynamo] graph break on isinstance calls if we don't know the type"


If we can't figure out the python type of a VariableTracker, then the
isinstance call should graph break (instead of raising an error).

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "[Dynamo] graph break on isinstance calls if we don't know the type"


If we can't figure out the python type of a VariableTracker, then the
isinstance call should graph break (instead of raising an error).

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 committed:
Update on "[Dynamo] graph break on isinstance calls if we don't know the type"


If we can't figure out the python type of a VariableTracker, then the
isinstance call should graph break (instead of raising an error).

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 commented:
Test failures are me needing to update dynamo_test_failures.py (RIP)

ydwu4 commented on torch/_dynamo/variables/builtin.py:
```diff
@@ -1053,7 +1053,12 @@ def call_getitem(self, tx, *args, **kwargs):
+            arg_type = arg.python_type()
+        except NotImplementedError:
+            unimplemented(
+                "isinstance({arg}, {isinstance_type}): can't determine type of {arg}"
```
nit: missing a `f` prefix?

zou3519 replied:
thanks, good catch

zou3519 committed:
Update on "[Dynamo] graph break on isinstance calls if we don't know the type"


If we can't figure out the python type of a VariableTracker, then the
isinstance call should graph break (instead of raising an error).

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 aakhundov kadeng

[ghstack-poisoned]

zou3519 commented:
@pytorchbot merge

zou3519 commented:
@pytorchbot merge -f "jobs are stuck?"
